src/Desktop.Analyzers/Core/TypesShouldNotExtendCertainBaseTypes.cs

```diff
@@ -31,79 +34,118 @@ public abstract class TypesShouldNotExtendCertainBaseTypesAnalyzer : DiagnosticA
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182171.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(SystemXmlXmlDocumentRule, SystemApplicationExceptionRule, SystemCollectionsCollectionBaseRule, SystemCollectionsDictionaryBaseRule, SystemCollectionsQueueRule, SystemCollectionsReadOnlyCollectionBaseRule, SystemCollectionsSortedListRule, SystemCollectionsStackRule);
 
+        private static readonly Dictionary<string, DiagnosticDescriptor> s_badBaseTypesToDescriptor = new Dictionary<string, DiagnosticDescriptor>
+                                                    {
+                                                        {"System.ApplicationException", SystemApplicationExceptionRule},
+                                                        {"System.Xml.XmlDocument", SystemXmlXmlDocumentRule},
+                                                        {"System.Collections.CollectionBase", SystemCollectionsCollectionBaseRule},
+                                                        {"System.Collections.DictionaryBase", SystemCollectionsDictionaryBaseRule},
+                                                        {"System.Collections.Queue", SystemCollectionsQueueRule},
+                                                        {"System.Collections.ReadOnlyCollectionBase", SystemCollectionsReadOnlyCollectionBaseRule},
+                                                        {"System.Collections.SortedList", SystemCollectionsSortedListRule},
+                                                        {"System.Collections.Stack", SystemCollectionsStackRule},
+                                                    };
+
         public override void Initialize(AnalysisContext analysisContext)
         {
-            
+            analysisContext.RegisterCompilationStartAction(AnalyzeCompilationStart);
+        }
+
+        private void AnalyzeCompilationStart(CompilationStartAnalysisContext context)
+        {
+            var badBaseTypes = s_badBaseTypesToDescriptor.Keys
+                                .Select(bt => context.Compilation.GetTypeByMetadataName(bt))
+                                .Where(bt => bt != null)
+                                .ToList();
+
+            context.RegisterSymbolAction((saContext) =>
+                {
+                    var namedTypeSymbol = saContext.Symbol as INamedTypeSymbol;
+                    if (!namedTypeSymbol.Locations.Any(l => l.IsInSource))
+                    {
+                        return;
+                    }
+
+                    var containedBadBaseTypes = badBaseTypes.Where(bbt => bbt.Equals(namedTypeSymbol.BaseType));
```

Comment:
In C# and VB, a type can extend only one base type. This is a language restriction. So, to answer your question, this can never be greater than 1.